src/components/homepage/HomeNavBar.js

```diff
@@ -1,85 +1,72 @@
-import React from "react";
-import { Navbar, Nav, Button, Container } from "react-bootstrap/";
-
-import { Link } from "react-router-dom";
+import React, { useState } from "react";
+import { Button } from "react-bootstrap";
 import { ReactComponent as FlowerHeartLogoMobile } from "../../assets/svgComponents/Icons/FlowerHeartLogoMobile.svg";
+import MenuOpen from "../../assets/svgComponents/MenuOpen";
+import MenuClose from "../../assets/svgComponents/MenuClose";
+import NavOverlay from "../NavOverlay";
+import { Link } from "react-router-dom";
+const token = localStorage.getItem("token");
 
 export default function HomeNavBar(props) {
-  const token = localStorage.getItem("token");
 
-  const handleLogout = () => props.onLogout();
+  const [open, setOpen] = useState(false);
 
+  //toggle's App 'showNav' state value, and local useState value
+  const toggle = () => {
+    console.log("clicked");
+    props.toggleClassOnNavOpen();
+    setOpen(!open);
```

Comment:
Another place to put a debugger would be here, and make sure everything is happening in the right order.

Also, is `open` a boolean? A more clear variable name might be `showNav` or `isOpen` or `navIsOpen`.